lib/fluent/plugin/in_systemd.rb

```diff
@@ -29,6 +29,7 @@ def start
       @running = true
       pos_writer.start
       @thread = Thread.new(&method(:run))
+      @thread.abort_on_exception = true
```

Comment:
Yep definitely agree that we shouldn't set it globally. I tried to set the thread-local flag here (http://ruby-doc.org/core-1.9.3/Thread.html#method-i-abort_on_exception) but I don't know Ruby well enough to know that I succeeded. Does that look like the right way to call the instance method rather than the class method?